code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm

```diff
@@ -1,47 +1,79 @@
 
 // ***************************************
-// *********** Sting
+// *********** Defile
 // ***************************************
-/datum/action/xeno_action/activable/larval_growth_sting/defiler
+/datum/action/xeno_action/activable/defile
 	name = "Defile"
 	action_icon_state = "defiler_sting"
-	mechanics_text = "Channel to inject an adjacent target with larval growth serum. At the end of the channel your target will be infected."
+	mechanics_text = "Injects an adjacent target with larval accelerant and a larva after a short wind up. All Defiler rapidly increase larval growth while the target has larval accelerant."
 	ability_name = "defiler sting"
 	plasma_cost = 150
-	cooldown_timer = 20 SECONDS
+	cooldown_timer = 60 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_DEFILE
 
-/datum/action/xeno_action/activable/larval_growth_sting/defiler/on_cooldown_finish()
-	playsound(owner.loc, 'sound/voice/alien_drool1.ogg', 50, 1)
-	to_chat(owner, "<span class='xenodanger'>You feel your toxin glands refill, another young one ready for implantation. You can use Defile again.</span>")
+/datum/action/xeno_action/activable/defile/on_cooldown_finish()
+	owner.playsound_local(owner, 'sound/effects/xeno_newlarva.ogg', 25, 0, 1)
+	to_chat(owner, "<span class='xenodanger'>You feel your toxin glands refill and another young one ready for implantation. You can use Defile again.</span>")
 	return ..()
 
-/datum/action/xeno_action/activable/larval_growth_sting/defiler/use_ability(atom/A)
+/datum/action/xeno_action/activable/defile/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return
+
+	if(!iscarbon(A))
+		if(!silent)
+			to_chat(owner, "<span class='xenodanger'>Our sting won't affect this target!</span>")
+		return FALSE
+
+	var/mob/living/carbon/victim = A
+
+	if(!A?.can_sting())
+		if(!silent)
+			to_chat(owner, "<span class='xenodanger'>Our sting won't affect this target!</span>")
+		return FALSE
+
+	if(CHECK_BITFIELD(victim.status_flags, XENO_HOST)) // already got one, stops doubling up
+		if(!silent)
+			to_chat(owner, "<span class='xenodanger'>This target already has a young one!</span>")
+		return FALSE
+
+	if(!owner.Adjacent(A))
+		var/mob/living/carbon/xenomorph/X = owner
+		if(!silent && world.time > (X.recent_notice + X.notice_delay))
+			to_chat(X, "<span class='warning'>We can't reach this target!</span>")
+			X.recent_notice = world.time //anti-notice spam
```

Comment:
Use the macros

Reply:
This is legacy code, not mine. TBH I don't even think there needs to be a delay